openmetadata-ui/src/main/resources/ui/src/components/common/entityPageInfo/EntityPageInfo.tsx

```diff
@@ -214,10 +226,13 @@ const EntityPageInfo = ({
                   <button
                     className={classNames(
                       'tw-text-xs tw-border-r tw-font-medium tw-py-1 tw-px-2 tw-rounded-l focus:tw-outline-none',
-                      isFollowing ? 'tw-border-white' : 'tw-border-primary'
+                      isFollowing ? 'tw-border-white' : 'tw-border-primary',
+                      { 'tw-cursor-not-allowed': deleted }
                     )}
                     data-testid="follow-button"
-                    onClick={followHandler}>
+                    onClick={() => {
+                      !deleted && followHandler && followHandler();
```

Comment:
```suggestion
                      !deleted && followHandler?.();
```
optional chaining makes calling optional functions easier than nullish coalescing.